operations/agent-flow-mixin/dashboards/opentelemetry.libsonnet

```diff
@@ -0,0 +1,174 @@
+  fieldConfig+: {
+    defaults+: {
+      custom+: {
+        fillOpacity: 20,
+        gradientMode: 'hue',
+        stacking: { mode: 'normal' },
+      },
+    },
+  },
+};
+
+{
+  [filename]:
+    dashboard.new(name='Grafana Agent Flow / OpenTelemetry') +
+    dashboard.withDashboardsLink() +
+    dashboard.withUID(std.md5(filename)) +
+    dashboard.withTemplateVariablesMixin([
+      dashboard.newTemplateVariable('cluster', |||
+        label_values(agent_component_controller_running_components, cluster)
+      |||),
+      dashboard.newTemplateVariable('namespace', |||
+        label_values(agent_component_controller_running_components{cluster="$cluster"}, namespace)
+      |||),
+      dashboard.newMultiTemplateVariable('instance', |||
+        label_values(agent_component_controller_running_components{cluster="$cluster", namespace="$namespace"}, instance)
+      |||),
+    ]) +
+    dashboard.withPanelsMixin([
+      // "Receivers for traces" row
+      (
+        panel.new('Receivers for traces [otelcol.receiver]', 'row') +
+        panel.withPosition({ h: 1, w: 24, x: 0, y: 0 })
+      ),
+      (
+        panel.new(title='Accepted spans', type='timeseries') +
+        panel.withDescription(|||
+         Number of spans successfully pushed into the pipeline.
+       |||) +
+        panel.withPosition({ x: 0, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr=|||
+              rate(receiver_accepted_spans_ratio_total{cluster="$cluster", namespace="$namespace", instance=~"$instance"}[$__rate_interval])
+            |||,
+            //TODO: How will the dashboard look if there is more than one receiver component? The legend is not unique enough?
+            legendFormat='{{ pod }} / {{ transport }}',
+          ),
+        ])
+      ),
+      (
+        panel.new(title='Refused spans', type='timeseries') +
+        stackedPanelMixin +
+        panel.withDescription(|||
+          Number of spans that could not be pushed into the pipeline.
+        |||) +
+        panel.withPosition({ x: 8, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr=|||
+             rate(receiver_refused_spans_ratio_total{cluster="$cluster", namespace="$namespace", instance=~"$instance"}[$__rate_interval])
+            |||,
+            legendFormat='{{ pod }} / {{ transport }}',
+          ),
+        ])
+      ),
+      (
+        panel.newHeatmap('RPC server duration (traces)') +
+        panel.withUnit('milliseconds') +
+        panel.withDescription(|||
+          The duration of inbound RPCs.
+        |||) +
+        panel.withPosition({ x: 16, y: 0, w: 8, h: 10 }) +
+        panel.withQueries([
+          panel.newQuery(
+            expr='sum by (le) (increase(rpc_server_duration_milliseconds_bucket{cluster="$cluster", namespace="$namespace", instance=~"$instance", rpc_service="opentelemetry.proto.collector.trace.v1.TraceService"}[$__rate_interval]))',
```

Comment:
I'm not very experience with Prometheus - can I ask someone to please double check that this PromQL makes sense?

Also, for this heatmap sometimes there are negative values in the most recent buckets:
![Screenshot 2024-01-17 at 16 28 53](https://github.com/grafana/agent/assets/5834788/13070546-397e-478a-834a-731f2f258f4d)

The negative values disappear once the bucket is older. I don't know why this happens?

Reply:
I am not very confident in the PromQL query because I copied it from another place in this mixin. I can understand the non-histogram queries in my dashboards, but this one about histograms is a bit over my head.